pkg/users/git.go

```diff
@@ -85,6 +85,8 @@ func (r *GitUserResolver) Resolve(user *gits.GitUser) (*jenkinsv1.User, error) {
 				}
 			}
 		}
+
+		gitUser.Login = naming.ToValidName(gitUser.Login)
```

Comment:
why the assignment to `gitUser.Login` and then a bit later to `id`? Is that not a bit confusing? Would it not work by just calling:
```
id = naming.ToValidName(gitUser.Login)
```
The is no need to change the `Login` member of `gitUser`, is it?

Reply:
The code here is very confusing you're right, sometimes we use the `gitUser.Login` and sometimes the `id`. But we always want the `gitUser.Login` to be lowercased before we try to create the CRD. That happens over here and we pass the `gitUser.Login` there. I've refactored the code but I'm not happy with the way things look atm. 

Let me know what you think :)